libgstc/libgstc_socket.c

```diff
@@ -72,7 +82,10 @@ gstc_socket_new (const char *address, const unsigned int port,
     goto out;
```

Comment:
Aren't you leaking `self` with this jump?

Reply:
no, since this point is only reached when the `self` pointer is `NULL` after attempting a `malloc`, this happens if it's unable to allocate the space